core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -141,7 +141,24 @@ public static ManifestReader<DataFile> read(
    * @return a manifest writer
    */
   public static ManifestWriter<DataFile> write(PartitionSpec spec, OutputFile outputFile) {
-    return write(1, spec, outputFile, null);
+    return write(spec, outputFile, /* compressionCodec */ null, /* compressionLevel */ null);
+  }
+
+  /**
+   * Create a new {@link ManifestWriter}.
+   *
+   * <p>Manifests created by this writer have all entry snapshot IDs set to null. All entries will
+   * inherit the snapshot ID that will be assigned to the manifest on commit.
+   *
+   * @param spec {@link PartitionSpec} used to produce {@link DataFile} partition tuples
+   * @param outputFile the destination file location
+   * @param compressionCodec compression codec for the manifest file
+   * @param compressionLevel compression level of the compressionCodec
+   * @return a manifest writer
+   */
+  public static ManifestWriter<DataFile> write(
```

Comment:
@sumeetgajjar, this must explicitly set the format version. This is a new method, so the format version should always be included. Older methods did not use a format version.

Reply:
Adding `formatVersion` here would make its signature the same as the other newly added `write` method. Thus removing it altogether.

Reply:
Removed the method altogether due to the reason stated above.